src/Sulu/Bundle/RouteBundle/Resources/js/containers/Form/fields/PageTreeRoute.js

```diff
@@ -0,0 +1,175 @@
+// @flow
+import React from 'react';
+import type {FieldTypeProps} from 'sulu-admin-bundle/types';
+import {observer} from 'mobx-react';
+import {action, observable} from 'mobx';
+import type {IObservableValue} from 'mobx';
+import {userStore} from 'sulu-admin-bundle/stores';
+import {Grid, ResourceLocator} from 'sulu-admin-bundle/components';
+import {SingleSelection, ResourceLocatorHistory} from 'sulu-admin-bundle/containers';
+import {translate} from 'sulu-admin-bundle/utils';
+import type {PageTreeRouteValue} from '../../types.js';
+import pageTreeRouteStyles from './pageTreeRoute.scss';
+
+type Props = FieldTypeProps<?PageTreeRouteValue>;
+
+@observer
+class PageTreeRoute extends React.Component<Props> {
+    @observable mode: string;
+
+    constructor(props: Props): void {
+        super(props);
+
+        const {
+            fieldTypeOptions: {
+                modeResolver,
+            },
+        } = props;
+
+        if (!modeResolver) {
+            throw new Error('The "modeResolver" must be a function returning a promise with the desired mode');
+        }
+
+        modeResolver(props).then(action((mode) => this.mode = mode));
```

Comment:
Looks like you have ignored my other comment :wink: This is some of the code that is duplicated from the `ResourceLocator` field type... Doesn't it make sense to reuse that field type here?